modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -6781,13 +6780,17 @@ else if (req.changeTopology()) {
                                         "previous [" + previous + "] with timeout " + backwardCheckTimeout);
                                 }
 
-                                liveAddr = checkConnection(new ArrayList<>(nodeAddrs), backwardCheckTimeout);
+                                liveAddr = checkConnection(previous, backwardCheckTimeout);
 
-                                if (log.isInfoEnabled()) {
-                                    log.info("Connection check to previous node done: [liveAddr=" + liveAddr
-                                        + ", previousNode=" + U.toShortString(previous) + ", addressesToCheck=" +
-                                        nodeAddrs + ", connectingNodeId=" + nodeId + ']');
-                                }
+                                String logMsg = "Connection check to previous node " + (liveAddr == null ? "failed" : "done")
+                                    + ". ConnectingNodeId=" + nodeId + ". PreviousNode=" + U.toShortString(previous)
+                                    + ", firstAliveAddr=" + liveAddr + ", addressesToCheck=" +
+                                    spi.getEffectiveNodeAddresses(previous) + "].";
```

Comment:
Suggestion: we could move the log into checkConnection(..) and reuse already called `spi.getEffectiveNodeAddresses(...)`.